airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -378,10 +376,16 @@ private SyncCheckConnectionFailure checkConnections(final GenerateInputActivity.
     return checkFailure;
   }
 
+  private boolean isResetJob(final IntegrationLauncherConfig sourceLauncherConfig) {
+    return sourceLauncherConfig.getDockerImage().equals(WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB);
```

Comment:
Consider reversing this to avoid a potential NPE:

```
WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB.equals(sourceLauncherConfig.getDockerImage())
```